hircluster.c

```diff
@@ -1016,6 +1029,7 @@ dict *parse_cluster_nodes(redisClusterContext *cc, char *str, int str_len,
                               sdsnewlen(master->addr, sdslen(master->addr)),
                               master);
                 if (ret != DICT_OK) {
+                    // Key already exists, but possibly an OOM error
```

Comment:
Is there no way to tell them apart? Can't we make dictAdd() return specific error codes such as DICT_OOM?

Reply:
its possible, would have to make sure the internals in dict handles it correctly for every case. Can do it in another pr